src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -456,97 +444,105 @@ private static void PerformInjection()
             return (Func<object[], T>)setterMethod.CreateDelegate(typeof(Func<object[], T>));
         }
 
-        /// <summary>
-        /// Initializes all the reflection objects it will ever need. Reflection is costly, but it's better to take
-        /// this one time performance hit than to get it multiple times later, or do it lazily and have to worry about
-        /// threading issues. If Initialize has been called before, it will not doing anything.
-        /// </summary>
-        private void Initialize()
-        {
-            lock (this)
-            {
-                if (!this.initialized)
-                {
-                    try
-                    {
-                        // This flag makes sure we only do this once. Even if we failed to initialize in an
-                        // earlier time, we should not retry because this initialization is not cheap and
-                        // the likelihood it will succeed the second time is very small.
-                        this.initialized = true;
-
-                        PrepareReflectionObjects();
-                        PerformInjection();
-                    }
-                    catch (Exception ex)
-                    {
-                        // If anything went wrong, just no-op. Write an event so at least we can find out.
-                        this.Write(InitializationFailed, new { Exception = ex });
-                    }
-                }
-            }
-        }
-
         private void RaiseRequestEvent(HttpWebRequest request)
         {
-            if (this.IsRequestInstrumented(request))
+            if (!ActivitySource.HasListeners() || this.IsRequestInstrumented(request))
             {
-                // This request was instrumented by previous RaiseRequestEvent, such is the case with redirect responses where the same request is sent again.
+                // No subscribers to the ActivitySource or this request was instrumented by previous
+                // RaiseRequestEvent, such is the case with redirect responses where the same request is sent again.
                 return;
             }
 
-            if (this.IsEnabled(ActivityName, request))
-            {
-                // We don't call StartActivity here because it will fire into user code before the headers are added.
+            var activity = ActivitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                Activity.Current?.Context ?? default,
```

Comment:
We shouldn't need to pass Activity.Current.context here.. It should be implicit.
We should pass context, only if we extracted context from remote.

I believe you are passing it because the StartActivity method expects context as well, when tags are to be passed at creation time?

Reply:
I could be wrong here, so asking @tarekgh for expert opinion :)

Reply:
Yes I ran into a bit of a "chicken-and-egg" situation here. I wanted tags to be present when Activity is started so anyone listening would have some data to make more intelligent sampling decisions. But ActivitySource API only gives me .StartActivity operations. No way to create Activity, build it up, and then Start it. So I tried to pass in everything I could on the StartActivity call which required then the context.

@tarekgh Any guidance to offer? What should I pass for context here?

Reply:
I believe you can pass just `default` here. That will cause the Activity created and the parent context will automatically picked from Activity.Current.

---
In reply to: [429364884](https://github.com/open-telemetry/opentelemetry-dotnet/pull/694#discussion_r429364884) [](ancestors = 429364884)